assets/js/redirect.js

```diff
@@ -0,0 +1,27 @@
+( function( $ ) {
+	$( function() {
+		const publishBtn = $('#publish');
+		const fromRule = $('#srm_redirect_rule_from');
+
+		fromRule.change(function(el) {
+			publishBtn.prop('disabled', true);
+			$.get(
+				window.ajaxurl,
+				{
+					action: 'srm_validate_from_url',
+					from: fromRule.val(),
+					_wpnonce: $('#srm_redirect_nonce').val()
+				}
+			).done(function( data ) {
+				if ( '1' === data ) {
+					$('#message').html( '' ).hide();
+					publishBtn.prop('disabled', false);
+				} else if ( '0' === data ) {
+					$('#message').html( '<p>There are some issues validating the URL. Please try again.</p>' ).show();
```

Comment:
@dhanendran Let's use localized strings exposed via [wp_localize_script](https://developer.wordpress.org/reference/functions/wp_localize_script/) instead of hardcoding these here and without any translations.